lib/src/store.dart

```diff
@@ -17,6 +17,9 @@ class Store {
   Pointer<Void> _cStore;
   final ModelDefinition defs;
 
+  // a list of observers of the Store.close() event
+  final Map<dynamic, void Function()> _onBeforeClose;
+
```

Comment:
@greenrobot-team & @Buggaboo please have a look at this functionality (including the remaining changes in `store.dart`). I'd like to cherry-pick that into main as it's also useful for other parts of the library, e.g. observable query `unsubscribe()` could be done automatically during store `close()`.

Reply:
cool!

Reply:
I don't follow what
```
listeners.forEach((observer) => observer());
```
is supposed to do. Am I missing something?

Reply:
executes each "onBeforeClose" event listeners, or in other words, issues the event to all listeners

Reply:
🤦 Ah, got it. I assumed this was supposed to call a close method that all observers should implement. Should have looked more closely at the type of `_onBeforeClose`... Observers will add themselves as observers to the Store close event.

LGTM!

Reply:
modified that calling code slightly and added some comments to improve readability

Reply:
I've changed this a little before merging and also took the liberty to update `observable.dart`. If you could, @Buggaboo, please have a look at 19b8008bf9269a9fa049de6952bbaeaeb1ad6aa5 if you see any issues

Reply:
Looks good!